RTNeural/common.h

```diff
@@ -146,11 +194,35 @@ static inline void sigmoid(const double* in, double* out, size_t dim) noexcept
     vvrec(out, out, &dim_int);
 }
 
+static inline void softmax(const float* in, float* out, size_t dim) noexcept
+{
+    constexpr float one = 1.0f;
+    const auto dim_int = static_cast<int>(dim);
+    float exp_sum;
+
+    vvexpf(out, in, &dim_int);
+    vDSP_sve(out, 1, &exp_sum, dim);
+    vDSP_vsdiv(out, 1, &exp_sum, out, 1, dim);
+}
+
+static inline void softmax(const double* in, double* out, size_t dim) noexcept
+{
+    constexpr double one = 1.0;
+    constexpr double neg_one = -1.0;
+    const auto dim_int = static_cast<int>(dim);
+    double exp_sum;
+
+    vvexp(out, in, &dim_int);
+    vDSP_sveD(out, 1, &exp_sum, dim);
+    vDSP_vsdivD(out, 1, &exp_sum, out, 1, dim);
+}
+
 } // namespace RTNeural
 
 #else // STL backend
 #include <algorithm>
 #include <cmath>
+#include <functional>
```

Comment:
Do we still need to include `<functional>`?

Reply:
Removed.